jenkinsfile-runner/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM maven:3.5.2 as maven
+
+RUN git clone https://github.com/jenkinsci/jenkinsfile-runner && \
```

Comment:
The Dockerfile for the jenkinsfile-runner uses a multi stage build to cache the mvn dependencies.